src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ADRController.kt

```diff
@@ -213,8 +225,64 @@ class ADRController(private val encryption: Encryption,
                 else -> adr.postFile("resource_patch", commonParameters + listOf("id" to resourceId), filePart)
             }
         }
-        finally {
+        finally
+        {
             tmpDir.deleteRecursively()
         }
     }
+
+    @Suppress("ReturnCount")
+    private fun makeOutputFileToPushToADR(artefact: Pair<ResponseEntity<StreamingResponseBody>, String>,
+                                          tmpDir: File,
+                                          resourceFileName: String): Pair<File?, ErrorDetail?>
```

Comment:
This method returns multiple values. I wonder if this could just be an Object instead of having to use Pair with null value.

Reply:
Could make a new type just for the return type of these methods, with file and errorDetail props, which would shut up the linting error, but feels like a bit of heavy overhead itself. Don't really mind either way, they're both kind of icky.